File: flutter_app/lib/models/appointment.dart
class Appointment {
  final String id;
  final String title;
  final String date;
  final String time;
  final String location;
  final String notes;
  final String type;
  final String status;
  final String? actionLabel;

  const Appointment({
    required this.id,
    required this.title,
    required this.date,
    required this.time,
    this.location = '',
    this.notes = '',
    this.type = 'appointment',
    this.status = 'todo',
    this.actionLabel,
  });

  Appointment copyWith({
    String? id,
    String? title,
    String? date,
    String? time,
    String? location,
    String? notes,
    String? type,
    String? status,
    String? actionLabel,
  }) {
    return Appointment(
      id: id ?? this.id,
      title: title ?? this.title,
      date: date ?? this.date,
      time: time ?? this.time,
      location: location ?? this.location,
      notes: notes ?? this.notes,
      type: type ?? this.type,
      status: status ?? this.status,
      actionLabel: actionLabel ?? this.actionLabel,
    );
  }

  @override
  String toString() => 'Appointment($title, $time, $status)';
}

File: react-native-app/src/models/Appointment.js
export function createAppointment({
  id,
  title,
  date,
  time,
  location = '',
  notes = '',
  type = 'appointment',
  status = 'todo',
  actionLabel = null,
}) {
  return { id, title, date, time, location, notes, type, status, actionLabel };
}

export function copyAppointment(appointment, overrides = {}) {
  return { ...appointment, ...overrides };
}
